tests/test-st2-invalid-auth.js

```diff
@@ -0,0 +1,167 @@
+      expect(error_spy).to.have.been.calledOnce;
+      expect(error_spy.firstCall.args[0]).include('Environment variables ST2_AUTH_USERNAME, ST2_AUTH_PASSWORD and ST2_AUTH_URL should only be used together.');
+      stop({shutdown: true});
+
+      done();
+    }).catch(function (err) {
+      console.log(err);
+      done(err);
+    });
+  });
+
+  it("should raise exception with bad auth URL", function(done) {
+    restore_env = mockedEnv({
+      ST2_AUTH_URL: 'https://nonexistent-st2-auth-url:9101',
+      ST2_AUTH_USERNAME: 'nonexistent-st2-auth-username',
+      ST2_AUTH_PASSWORD: 'nonexistent-st2-auth-password'
+    });
+
+    // Load script under test
+    var i, stackstorm = require("../scripts/stackstorm.js");
+    stackstorm(robot).then(function (stop) {
+      expect(error_spy.args).to.have.lengthOf.above(2);
+
+      // Check that it was called at some point with 'Failed to retrieve commands from'
+      for (i = 0; i < error_spy.args.length; i++) {
+        try {
+          expect(error_spy.args[i][0]).to.be.a('string');
+          expect(error_spy.args[i][0]).to.startWith('Failed to authenticate with st2 username and password:');
+          break;
+        } catch (err) {
+          // If we have reached the last call and we still haven't found it
+          if (i >= error_spy.args.length-1) {
+            // Re-throw the assert exception
+            throw(err);
+          }
+          // Implicit continue
+        }
+      }
+
+      // Check that it was called at some point with 'Failed to retrieve commands from'
+      for (i = 0; i < error_spy.args.length; i++) {
+        try {
+          expect(error_spy.args[i][0]).to.be.a('string');
+          expect(error_spy.args[i][0]).to.include('Failed to authenticate with st2 username and password');
+          break;
+        } catch (err) {
+          // If we have reached the last call and we still haven't found it
+          if (i >= error_spy.args.length-1) {
+            // Re-throw the assert exception
+            throw(err);
+          }
+          // Implicit continue
+        }
+      }
+
+      stop({shutdown: true});
+
+      done();
+    }).catch(function (err) {
+      console.log(err);
+      done(err);
+    });
+  });
+
+  it("should raise exception with bad API key", function(done) {
+    restore_env = mockedEnv({
+      ST2_API_KEY: 'aaa'
+    });
+
+    // Load script under test
+    var stackstorm = require("../scripts/stackstorm.js");
+    stackstorm(robot).then(function (stop) {
+      expect(info_spy).to.have.been.calledTwice;
+      expect(info_spy).to.have.been.calledWith('Using ST2_API_KEY as authentication. Expiry will lead to bot exit.');
+      expect(info_spy).to.have.been.calledWith('Loading commands...');
+
+      stop({shutdown: true});
+
+      done();
+    }).catch(function (err) {
+      console.log(err);
+      done(err);
+    });
+  });
+
+  it("should raise exception with bad auth token", function(done) {
```

Comment:
@jinpingh We really needs these 2 tests to be helpful.

From what I see current test promises it "should raise exception with bad auth token", however what it really does is checking for "Loading commands" message in the log, without verifying if exception was raised or bot is terminated.

Please make sure these tests actually test that hubot is called to exit on invalid ST2 token or key.

Reply:
Modified test cases for token and API key yesterday. It ran on my two different systems successfully but failed on Travis. Removed changes and will work on that.

Reply:
Cool, thanks!